modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java

```diff
@@ -190,13 +205,27 @@ public FriendlyURLEntry deleteFriendlyURLEntry(
 			}
 			else {
 				friendlyURLEntryMapping.setFriendlyURLEntryId(
-					friendlyURLEntry.getFriendlyURLEntryId());
+					friendlyURLEntryId);
 
 				_friendlyURLEntryMappingPersistence.update(
 					friendlyURLEntryMapping);
 			}
 		}
 
+		// Asset
+
+		try {
+			AssetEntry assetEntry = _assetEntryLocalService.fetchEntry(
+				FriendlyURLEntry.class.getName(), friendlyURLEntryId);
+
+			if (assetEntry != null) {
+				_assetEntryLocalService.deleteEntry(assetEntry.getEntryId());
+			}
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
```

Comment:
@adolfopa, unsure if we should throw a RuntimeException here or just log it. What do you think?

Reply:
I think this should fail and bubble up the error, unless there's a very good reason not to.